container-core/src/main/java/com/yahoo/metrics/simple/Point.java

```diff
@@ -58,32 +60,16 @@ private static Tuple2<String[], Value[]> buildParameters(Map<String, ?> properti
 
     @Override
     public boolean equals(Object obj) {
-        if (this == obj) {
-            return true;
-        }
-        if (obj == null) {
-            return false;
-        }
-        if (getClass() != obj.getClass()) {
-            return false;
-        }
+        if (this == obj) return true;
+        if (obj == null) return false;
+        if (getClass() != obj.getClass()) return false;
         Point other = (Point) obj;
-        if (!Arrays.equals(dimensions, other.dimensions)) {
-            return false;
-        }
-        if (!Arrays.equals(location, other.location)) {
-            return false;
-        }
-        return true;
+        return Arrays.equals(dimensions, other.dimensions) && Arrays.equals(location, other.location);
     }
 
     @Override
     public int hashCode() {
-        final int prime = 31;
-        int result = 1;
-        result = prime * result + Arrays.hashCode(dimensions);
-        result = prime * result + Arrays.hashCode(location);
-        return result;
+        return hashCode;
     }
 
```

Comment:
Consider using `List.of()` for immutable view for return value of `location()` and `dimensions()`. This may lead to a zero-copy optimization downstream.

Reply:
Done